src/index.ts

```diff
@@ -1,8 +1,52 @@
 import * as cookie from 'cookie'
 import * as next from 'next'
+import * as setCookieParser from 'set-cookie-parser'
+import { Cookie } from 'set-cookie-parser'
+import { CookieSerializeOptions } from 'cookie'
 
 const isBrowser = () => typeof window !== 'undefined'
 
+/**
+ * Compare the cookie and return true if the cookies has equivalent
+ * options and the cookies would be overwritten in the browser storage.
+ *
+ * @param a first Cookie for comparision
+ * @param b second Cookie for comparision
+ */
+function areCookiesEqual(a: Cookie, b: Cookie) {
+  return (
+    a.name !== b.name ||
+    a.domain !== b.domain ||
+    a.path !== b.path ||
+    a.httpOnly !== b.httpOnly ||
+    a.secure !== b.secure
+  )
+}
+
+/**
+ * Create an instance of the Cookie interface
+ *
+ * @param name name of the Cookie
+ * @param value value of the Cookie
+ * @param options Cookie options
+ */
+function createCookie(
```

Comment:
Could you elaborate on the idea behind `createCookie` function? I am not sure I entirely understand why we necessarily need it.

Reply:
This is just to make the code more clean. The function is not needed at all, but it makes the code below more readable and easier to understand and maintain from my point of view. If you don't want the function here I can move the code directly to the setCookie function.